ldk/go/readme.md

```diff
@@ -11,9 +11,76 @@ go get -u github.com/open-olive/loop-development-kit/ldk/go
 Full documentation available on [pkg.go.dev](https://pkg.go.dev/github.com/open-olive/loop-development-kit/ldk/go).
 
 ### Loops
-This LDK can be used to write loops for Olive Helps. More detail about loops is available [here](docs/loops.md).
+Loops receive events and use them to generate relevant whispers. Loops choose which events they want to use and which they want to ignore.
 
-Example loops are available in the examples folder.
+This LDK can be used to write loops for Olive Helps. An example of the loop implementation is available [here](docs/loops.md).
+
+
+#### Interface
+
+Writing a loop boils down to writing an implementation for the `Loop` interface and serving it via `ldk.ServeLoopPlugin`.
+
+```go
+type Loop interface {
+	LoopStart(Sidekick) error
+	LoopStop() error
+}
+
+func main() {
+	...
+	ldk.ServeLoopPlugin(logger, loop) // ServeLoopPlugin, gives an ability to provide your own logger
+}	
+```
+
+**LoopStart** - The Loop should wait to start operating until this is called. The provided `Sidekick` reference should be stored in memory for continued use.
+
+**LoopStop** - The Loop should stop operating when this is called.
+
+#### Subscribing to Sensors
+
+Inside `LoopStart`, you can subscribe to various sensors. Here's an example of subscribing to a couple:
+
+```go
+func (c *Loop) LoopStart(sidekick ldk.Sidekick) error {
+	// ...
+
+	handler := func (text string, err error) {
+		// Respond to sensor event here...
+	}
+
+	if err := sidekick.Clipboard().Listen(l.ctx, handler); err != nil {
+		return err
+	}
+
+	if err := sidekick.UI().ListenSearchbar(l.ctx, handler); err != nil {
+		return err
+	}
+}
+```
+
+Loops do not need to emit whispers in a 1:1 relationship with events. Loops may not use events at all. Loops may only use some events. Loops may keep a history of events and only emit whispers when several conditions are met.
+
+#### List of possible Sensors to subscribe to
```

Comment:
Now that I think about it, and taking to account the developer hub aptitude list we have available, maybe we should link to that instead...

Reply:
Not a bad idea. Do you have a link to where would we like to point them?

Reply:
Hmm... not a great link right now - but I like having the interface available instead.

Oops! one thing though, we need to replace references to "Sensors" with "Aptitudes" in the documentation, we don't call them Sensors any more so we should avoid them in the documents.

Reply:
Ahh, I did that and then changed it back based on the response in the slack. I'll change it again.

Reply:
Updated